doc/howto.md

```diff
@@ -63,3 +63,50 @@ FROM penguins.csv
 GROUP BY species
 ORDER BY count DESC
 ```
+
+## Register SQLite UDF
+
+To register a user-defined function (UDF) when using SQLite, you can use [SQLAlchemy's `@event.listens_for`](https://docs.sqlalchemy.org/en/14/dialects/sqlite.html#user-defined-functions) and SQLite's [`create_function`](https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.create_function):
+
+### Install JupySQL
+
+```{code-cell} ipython3
+%pip install jupysql --quiet
+```
+
+### Create engine and register function
+
+```{code-cell} ipython3
+from sqlalchemy import create_engine
+from sqlalchemy import event
+
+def mysum(x, y):
+    return x + y
+
+engine = create_engine("sqlite://")
+
+@event.listens_for(engine, "connect")
+def connect(conn, rec):
+    conn.create_function(name="MYSUM", narg=2, func=mysum)
+```
+
+### Create connection with existing engine
+
+```{versionadded} 0.5.1
```

Comment:
This is a good example. Can't we allow the functionality without passing existing engine? It adds complexity and friction to the users.

Reply:
we cannot. the engine has to be prepared beforehand to register the function. I'm unsure what you're proposing

Reply:
Can't you do something like that?

```python
from sqlalchemy import create_engine, func

# Define a function that takes in a value and returns its square
def my_square_function(x):
    return x * x

# Create an engine and connect to a database
engine = create_engine('postgresql://username:password@localhost:5432/mydatabase')
connection = engine.connect()

# Use the func module to define the UDF as a SQL expression
square_udf = func.my_square_function(column)

# Use the UDF in a SELECT statement
query = (
    'SELECT *, {} AS squared_value FROM mytable'.format(square_udf)
)
result = connection.execute(query)

# Print the result set
for row in result:
    print(row)

# Close the connection
connection.close()
In this example, we define a function my_square_function that takes in a value and returns its square. We then use the func module to create a SQL expression for the function, and use it in a SELECT statement.
```
*Keep in mind that this example uses PostgreSQL as the database backend, and you may need to modify the example slightly to work with a different database.

Reply:
can you provide an example of how you think the API should be when using `%%sql`? The snippet you shared is not using it

Reply:
```python
# Load the extension and connect to the database
%load_ext sql
%sql postgresql://username:password@localhost:5432/mydatabase

# Define a function that takes in a value and returns its square
def my_square_function(x):
    return x * x

# Use the func module to define the UDF as a SQL expression
from sqlalchemy import func
square_udf = func.my_square_function(column)

# Use the UDF in a SELECT statement
result = %sql SELECT *, :square_udf AS squared_value FROM mytable

# Print the result set
print(result)
```

We might want to even create a tighter integration to the func part, maybe something like:

```python
# Define the UDF as a SQL expression
%sql --udf my_square_function
```

Reply:
Ok, I see. I think we can open an issue and discuss the API, I agree it's simpler. For now, I'd say let's get this example in the docs so people know they can do it, then we simplify it.

btw, I don't think `sqlalchemy.func` is what we want, [based on this](https://docs.sqlalchemy.org/en/14/tutorial/data_select.html#tutorial-functions), it looks like it's module to construct SQL from Python, not really for UDFs.

Reply:
Alright, I'll keep the original issue open.

I think it's `func` with another thing [like select()](https://stackoverflow.com/questions/17597575/sqlalchemy-user-defined-function-error), the developer should dive into it.